src/devices/device_models/AC_branches.jl

```diff
@@ -26,89 +26,252 @@ function flow_variables(ps_m::CanonicalModel,
+        end
 
-    add_variable(ps_m,
-                 devices,
-                 time_steps,
-                 Symbol("Fbr_$(B)"),
-                 false)
+    end
 
     return
 
 end
 
+function flow_variables(ps_m::CanonicalModel,
+                        system_formulation::Type{S},
+                        devices::PSY.FlattenedVectorsIterator{B},
+                        time_steps::UnitRange{Int64}) where {B <: PSY.ACBranch,
+                                                             S <: PM.AbstractPowerFormulation}
+    #Get PowerModels dicts
+    pm_object = ps_m.pm_model
+    ref = pm_object.ref
+    var = pm_object.var
+    arc_ix = pm_object.ext[:arc_ix]
+
+    var_name_from_p = Symbol("Pbr_fr_$(B)")
+    var_name_to_p = Symbol("Pbr_to_$(B)")
+    var_name_from_q = Symbol("Qbr_fr_$(B)")
+    var_name_to_q = Symbol("Qbr_to_$(B)")
+
+    ps_m.variables[var_name_to_p] = PSI._container_spec(ps_m.JuMPmodel,
+                                                       (d.name for d in devices),
+                                                       time_steps)
+
+    ps_m.variables[var_name_from_p] = PSI._container_spec(ps_m.JuMPmodel,
+                                                          (d.name for d in devices),
+                                                          time_steps)
+
+
+    ps_m.variables[var_name_to_q] = PSI._container_spec(ps_m.JuMPmodel,
+                                                       (d.name for d in devices),
+                                                       time_steps)
+
+    ps_m.variables[var_name_from_q] = PSI._container_spec(ps_m.JuMPmodel,
+                                                          (d.name for d in devices),
+                                                          time_steps)
+
+
+    for (ix, d) in enumerate(devices)
+        arc_ix = arc_ix + 1
+        bus_fr = d.connectionpoints.from.number
+        bus_to = d.connectionpoints.to.number
+        arcs_fr = (arc_ix, bus_fr, bus_to)
+        arcs_to = (arc_ix, bus_to, bus_fr)
+
+        for t in time_steps
+            ref[:nw][t][:branch][arc_ix] = PSI.get_branch_to_pm(arc_ix, d)
+            push!(ref[:nw][t][:arcs_from], arcs_fr)
+            push!(ref[:nw][t][:arcs_to], arcs_to)
+            #careful here the order of the push has to match the order of the variable creation
+            push!(ref[:nw][t][:arcs], arcs_fr, arcs_to)
+
+            #Active Power Variables
+            ps_m.variables[var_name_from_p][d.name,t] = JuMP.@variable(ps_m.JuMPmodel,
+                                                                            base_name="$(bus_fr),$(bus_to)_{$(d.name),$(t)}",
+                                                                            upper_bound = d.rate,
+                                                                            lower_bound = -d.rate,
+                                                                            start = 0.0)
```

Comment:
Instead of creating these variables yourself, why not let PowerModels create them and extract them into your data structure?